internal/pkg/cli/pipeline_init.go

```diff
@@ -192,6 +201,49 @@ func (o *initPipelineOpts) Ask() error {
 	return nil
 }
 
+func (o *initPipelineOpts) validateDuplicatePipeline() error {
+	// make sure pipeline isn't already deployed
+	fullName := fmt.Sprintf(fmtPipelineName, o.appName, o.name)
+
+	_, err := o.pipeline.GetPipeline(fullName)
+	switch {
+	case err == nil:
+		log.Errorf(`It seems like you are trying to init a pipeline that already exists.
+To recreate the pipeline, please run:
+1. %s. Note: The manifest file will not be deleted and will be used in Step 2.
+If you'd prefer a new default manifest, please manually delete the existing file.
+2. And then %s
+`,
+			color.HighlightCode(fmt.Sprintf("copilot pipeline delete --name %s", o.name)),
+			color.HighlightCode(fmt.Sprintf("copilot pipeline init --name %s", o.name)))
+		return fmt.Errorf("pipeline %s already exists", color.HighlightUserInput(o.name))
+	case !errors.Is(err, codepipeline.ErrPipelineNotFound):
+		return fmt.Errorf("validate if pipeline exists: %w", err)
+	}
+
+	// make sure pipeline doesn't exist locally
```

Comment:
That's an interesting thought 🤔 . Currently, if a service manifest file gets deleted, we block `init`, but you're right, `pipeline init` only writes local files, so we could handle it a little different.

I think the potential options in your situation are:
1. Error out (current outcome)
2. Pull the current pipeline configuration and recreate that manifest (no log, may not be possible without storing some info in SSM)
3. Generate a fresh pipeline configuration (log)

I think the best approach is what's currently written - to block `pipeline init` if there is a pipeline that's already deployed with that name, but I could be swayed to either of the other options if you think another option would be better. Thoughts?

Reply:
OK it makes sense to me!

Reply:
I agree, bc if they are naming a new pipeline "repoName", it would not be in a local manifest (because "pipeline-appName-repoName" would), but would end up with a namespace clash upon deployment.

Reply:
Oh, nvm. Finally read to the bottom. 😆